code/modules/assembly/proximity.dm

```diff
@@ -114,49 +110,48 @@
 		holder.update_icon()
 	return
 
-/obj/item/assembly/prox_sensor/ui_interact(mob/user)//TODO: Change this to the wires thingy
-	. = ..()
+/obj/item/assembly/prox_sensor/ui_status(mob/user)
 	if(is_secured(user))
-		var/second = time % 60
-		var/minute = (time - second) / 60
-		var/dat = "<TT><B>Proximity Sensor</B></TT>"
-		if(!scanning)
-			dat += "<BR>[(timing ? "<A href='?src=[REF(src)];time=0'>Arming</A>" : "<A href='?src=[REF(src)];time=1'>Not Arming</A>")] [minute]:[second]"
-			dat += "<BR><A href='?src=[REF(src)];tp=-30'>-</A> <A href='?src=[REF(src)];tp=-1'>-</A> <A href='?src=[REF(src)];tp=1'>+</A> <A href='?src=[REF(src)];tp=30'>+</A>"
-		dat += "<BR><A href='?src=[REF(src)];scanning=[scanning?"0'>Armed":"1'>Unarmed (Movement sensor active when armed!)"]</A>"
-		dat += "<BR>Detection range: <A href='?src=[REF(src)];sense=down'>-</A> [sensitivity] <A href='?src=[REF(src)];sense=up'>+</A>"
-		dat += "<BR><BR><A href='?src=[REF(src)];refresh=1'>Refresh</A>"
-		dat += "<BR><BR><A href='?src=[REF(src)];close=1'>Close</A>"
-		user << browse(dat, "window=prox")
-		onclose(user, "prox")
-		return
-
-
-/obj/item/assembly/prox_sensor/Topic(href, href_list)
-	..()
-	if(!usr.canUseTopic(src, BE_CLOSE))
-		usr << browse(null, "window=prox")
-		onclose(usr, "prox")
-		return
-
-	if(href_list["sense"])
-		sensitivity_change(((href_list["sense"] == "up") ? 1 : -1))
-
-	if(href_list["scanning"])
-		toggle_scan(text2num(href_list["scanning"]))
-
-	if(href_list["time"])
-		timing = text2num(href_list["time"])
-		update_icon()
-
-	if(href_list["tp"])
-		var/tp = text2num(href_list["tp"])
-		time += tp
-		time = min(max(round(time), 0), 600)
-
-	if(href_list["close"])
-		usr << browse(null, "window=prox")
+		return ..()
+	return UI_CLOSE
+
+/obj/item/assembly/prox_sensor/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, \
+									datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "proximity_sensor", name, ui_x, ui_y, master_ui, state)
+		ui.open()
+
+/obj/item/assembly/prox_sensor/ui_data(mob/user)
+	var/list/data = list()
+	var/time_left = time
+	data["seconds"] = round(time_left % 60)
+	data["minutes"] = round((time_left - data["seconds"]) / 60)
+	data["timing"] = timing
+	data["scanning"] = scanning
+	data["sensitivity"] = sensitivity
+	return data
+
+/obj/item/assembly/prox_sensor/ui_act(action, params)
+	if(..())
 		return
 
-	if(usr)
-		attack_self(usr)
+	switch(action)
+		if("scanning")
+			var/toggle = scanning ? FALSE : TRUE
+			toggle_scan(toggle)

```

Comment:
Is this not just `toggle_scan(!scanning)`?

Reply:
Yep, you are right. Good catch.